portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -62,4 +69,32 @@ private String convertToJsonUsingGson(Message messages) {
         String json = gson.toJson(messages);
         return json;
     }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+        String text = getParameter(request, "text-input", "");
+
+    // Respond with the result.
+        response.setContentType("text/html;");
+        response.getWriter().println(text);
+
+    // Add comment to arraylist
+        sampleMessage.add(text);
+
+    // Redirect back to the HTML page.
+        response.sendRedirect("/index.html");
+    }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+        String value = request.getParameter(name);
+        if (value == null) {
```

Comment:
In Java, "==" compares whether the references are equal, while the equals() method compares whether the contents are actually equal.

I believe you would want to use equals() here:
  if (value.equals(""))...